stock_picking_inter_warehouse/models/stock_warehouse.py

```diff
@@ -0,0 +1,92 @@
+from odoo import _, fields, models
+
+
+class StockWarehouse(models.Model):
+    _inherit = "stock.warehouse"
+
+    inter_warehouse_transfers = fields.Boolean("Inter-Warehouse Transfers")
+
+    receipt_destination_location_id = fields.Many2one(
+        "stock.location", "Receipt Destination Location"
+    )
+    receipt_picking_type_id = fields.Many2one(
+        "stock.picking.type", "Receipt Operation Type"
+    )
+    receipt_picking_partner_id = fields.Many2one("res.partner", "Partner in Picking")
+
+    def write(self, vals):
+        res = super().write(vals)
+        for wh in self:
+            if wh.inter_warehouse_transfers:
+                company_internal_transit_location = (
+                    wh.company_id.internal_transit_location_id
+                )
+                if not company_internal_transit_location.active:
+                    company_internal_transit_location.write(
+                        {
+                            "active": True,
+                        }
+                    )
+
+                location_src_id = self.env["stock.location"].search(
+                    [
+                        ("name", "=", wh.name),
+                        ("location_id", "=", company_internal_transit_location.id),
+                    ]
+                )
+                if not location_src_id:
+                    location_src_id = self.env["stock.location"].create(
+                        {
+                            "name": wh.name,
+                            "location_id": company_internal_transit_location.id,
+                        }
+                    )
+                wh.receipt_picking_partner_id.default_stock_location_src_id = (
+                    location_src_id.id
+                )
+
+                route_inter_warehouse = self.env["stock.location.route"].search(
+                    [("name", "ilike", "Inter-Warehouse")]
+                )
```

Comment:
I think better to use barcode field (for create and search), since name is transalted.